src/abstraction.jl

```diff
@@ -3,9 +3,18 @@ module Abstraction
 using LinearAlgebra
 using ProgressMeter
 using StaticArrays
+using Base.Cartesian
 
 @enum INCL_MODE INNER OUTER
 
+Base.hash(x::Tuple{}, h::UInt) = h + Base.tuplehash_seed
```

Comment:
Reference the JuliaLang issue from here explaining why we need type piracy.
Type piracy is when you define a method for which the function is defined outside your package and the types in the signatures too. Here `hash`, `Tuple` and `UInt` are defined in `Base` so it is type piracy: loading Dionysos will change code that does not even use Dionysos.
It's ok if we reference the JuliaLang issue showing it is a performance hack